app/controllers/api/v1/bibs_controller.rb

```diff
@@ -1,31 +1,81 @@
 class Api::V1::BibsController < ApplicationController
+  include LogWrapper
+  # Receive teacher sets from a POST request.
+  # All records are inside @request_body.
+  # Find or create a teacher set in the MLN db and its associated books.
   def create_or_update_teacher_sets
-    render status: 200, json: {
-      teacher_sets: [
-        {
-          id: 1,
-          title: 'Example title 1'
-        },
-        {
-          id: 2,
-          title: 'Example title 2'
-        }
-      ]
-    }.to_json
+    return if !request_valid?('create_or_update_teacher_sets')
+
+    saved_teacher_sets = []
+    @request_body.each do |teacher_set_record|
+      teacher_set = TeacherSet.where(bnumber: "b#{teacher_set_record['id']}").first_or_initialize
+      teacher_set.update_attributes(
+        title: teacher_set_record[:title]
+        # TODO: update more teacher_set attributes here once the API contract is complete
+      )
+      teacher_set.update_included_book_list(teacher_set_record)
+      saved_teacher_sets << teacher_set if teacher_set.save
```

Comment:
what happens if .save fails?

Reply:
`.save` fails if the record fails validation, and therefore we shouldn't return it in the response body.

Reply:
But the value you return, is it a nil, or underfined?  Basically, is this a third type of possible return value?